Lib/tarfile.py

```diff
@@ -525,7 +525,7 @@ def read(self, size=None):
                 if not buf:
                     break
                 t.append(buf)
-            buf = "".join(t)
+            buf = b"".join(t)
```

Comment:
nice catch.

Reply:
It never caused a problem, since this line is never called;  `size` is never None in the function call. But still, should be fixed, I guess.